06_puppy/kasunfdo/syncstore.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+func (s *SyncStore) CreatePuppy(puppy *Puppy) bool {
+	_, ok := s.Map.LoadOrStore(puppy.ID, puppy)
```

Comment:
While I can see what this does, it took a little while to step through what was happening and why we were using LoadOrStore.

Potentially you could consider just doing a separate line that checks for existence by calling Load, and then calling plain old Store instead of LoadOrStore.

Reply:
Valid point. Initially I was using Store method with checking for existence. After going through the sync map reference, I thought of using existing implementation and changed Store to LoadOrStore and added a comment to make it much clear.